internals/plan/plan.go

```diff
@@ -1021,82 +1022,111 @@ func validServiceAction(action ServiceAction, additionalValid ...ServiceAction)
 
 var fnameExp = regexp.MustCompile("^([0-9]{3})-([a-z](?:-?[a-z0-9]){2,}).yaml$")
 
-func ReadLayersDir(dirname string) ([]*Layer, error) {
+type LayerReader struct {
```

Comment:
Seems like this should not be exported (`layerReader`).

Reply:
sounds good.